pandas/core/dtypes/common.py

```diff
@@ -731,11 +731,23 @@ def pandas_dtype(dtype):
             except TypeError:
                 pass
 
+        elif dtype == 'object':
+            return np.dtype(dtype)
+
         try:
             return CategoricalDtype.construct_from_string(dtype)
         except TypeError:
             pass
     elif isinstance(dtype, ExtensionDtype):
         return dtype
+    else:
+        try:
+            np.dtype(dtype)
+        except (TypeError, ValueError):
+            raise
+        if dtype in [object, np.object_, list]:
+            return np.dtype(dtype)
+        elif np.dtype(dtype).kind == 'O':
```

Comment:
@jreback From what I have understood, any invalid dtype (such as `pd.Timestamp`) should raise an error. `np.dtype(invalid_type).kind = 0` for such objects. However, this will also catch some valid `dtypes` such as `object, np.object_ and 'object'` which we safeguard against by catching them earlier and returning `np.dtype(valid_dtype)` in before this condition is evaluated.